src/main/java/com/checkmarx/intellij/inspections/AscaInspection.java

```diff
@@ -0,0 +1,193 @@
+            return "";
+        }
+        return text.replace("&", "&amp;")
+                .replace("<", "&lt;")
+                .replace(">", "&gt;")
+                .replace("\"", "&quot;")
+                .replace("'", "&#39;");
+    }
+
+    /**
+     * Gets the text range for a specific line in the document.
+     *
+     * @param document the document
+     * @param lineNumber the line number
+     * @return the text range
+     */
+    private TextRange getTextRangeForLine(Document document, int lineNumber) {
+        int startOffset = document.getLineStartOffset(lineNumber - 1);
+        int endOffset = Math.min(document.getLineEndOffset(lineNumber - 1), document.getTextLength());
+        return new TextRange(startOffset, endOffset);
+    }
+
+    /**
+     * Checks if the line number is out of range in the document.
+     *
+     * @param lineNumber the line number
+     * @param document the document
+     * @return true if the line number is out of range, false otherwise
+     */
+    private boolean isLineOutOfRange(int lineNumber, Document document) {
+        return lineNumber <= 0 || lineNumber > document.getLineCount();
+    }
+
+    /**
+     * Checks if the scan result is invalid.
+     *
+     * @param scanResult the scan result
+     * @return true if the scan result is invalid, false otherwise
+     */
+    private boolean isInvalidScan(ScanResult scanResult) {
+        return scanResult == null || scanResult.getScanDetails() == null;
+    }
+
+    /**
+     * Determines the highlight type for a specific scan detail.
+     *
+     * @param detail the scan detail
+     * @return the problem highlight type
+     */
+    private ProblemHighlightType determineHighlightType(ScanDetail detail) {
+        return getSeverityToHighlightMap().getOrDefault(detail.getSeverity(), ProblemHighlightType.WEAK_WARNING);
+    }
+
+    /**
+     * Gets the map of severity to highlight type.
+     *
+     * @return the map of severity to highlight type
+     */
+    private Map<String, ProblemHighlightType> getSeverityToHighlightMap() {
+        if (severityToHighlightMap == null) {
+            severityToHighlightMap = new HashMap<>();
+            severityToHighlightMap.put(Constants.ASCA_CRITICAL_SEVERITY, ProblemHighlightType.GENERIC_ERROR);
+            severityToHighlightMap.put(Constants.ASCA_HIGH_SEVERITY, ProblemHighlightType.GENERIC_ERROR);
+            severityToHighlightMap.put(Constants.ASCA_MEDIUM_SEVERITY, ProblemHighlightType.WARNING);
+            severityToHighlightMap.put(Constants.ASCA_LOW_SEVERITY, ProblemHighlightType.WEAK_WARNING);
+        }
+        return severityToHighlightMap;
+    }
+
+    /**
+     * Performs an ASCA scan on the given file.
+     *
+     * @param file the file to scan
+     * @return the scan result
+     */
+    private ScanResult performAscaScan(PsiFile file) {
+        return new AscaService().runAscaScan(file, file.getProject(), false, Constants.JET_BRAINS_AGENT_NAME);
```

Comment:
I added one AscaService, but we cannot create a new constructor with params, because intellij is trying to call the default one. I added setter and getter in case we will want to override the value by a mock value